### PR TITLE
Use component config rather that global model data config

### PR DESCRIPTION
**Pull request by dswiecki**
Merge to staging after `release/1.1` branch is cut

### CODE REVIEW

File: engine/flink/generic/src/main/resources/defaultModelConfig.conf [hunk 1]
```diff
@@ -7,19 +7,24 @@
...
   }
-
-  kafka {
-    kafkaAddress: ${?KAFKA_ADDRESS}
-    kafkaProperties {
-      "schema.registry.url": ${?SCHEMA_REGISTRY_URL}
+  components {
+    kafka {
+      config {
+        kafka {
```
Comment (mproch) [inline engine/flink/generic/src/main/resources/defaultModelConfig.conf:18]: do we need so much nesting?

Comment (dswiecki) [reply engine/flink/generic/src/main/resources/defaultModelConfig.conf]: right now its nested, but we discused it with @arkadius that config should already be kafka config

Comment (dswiecki) [reply engine/flink/generic/src/main/resources/defaultModelConfig.conf]: I've flattened the structure

File: engine/flink/generic/src/main/resources/defaultModelConfig.conf [hunk 2]
```diff
@@ -7,21 +7,12 @@
...
   timeout: 10s
   checkpointConfig {
-      checkpointInterval: 10m
+    checkpointInterval: 10m
   }
   rocksDB {
     enable: true
   }
 
-  kafka {
```
Comment (arkadius) [inline engine/flink/generic/src/main/resources/defaultModelConfig.conf:16]: Shouldn't be copied to `dev-application.conf` and to `src/universal/conf/application.conf`?

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/kryo/AvroSerializersRegistrar.scala [hunk 3]
```diff
@@ -16,27 +18,36 @@ class AvroSerializersRegistrar extends SerializersRegistrar with LazyLogging {
 
   override def register(modelConfig: Config, executionConfig: ExecutionConfig): Unit = {
     logger.debug("Registering default avro serializers")
+    val componentsConfig = modelConfig.getAs[Map[String, ComponentProviderConfig]]("components").getOrElse(Map.empty)
+    val componentKafkaConfig = componentsConfig
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/kryo/AvroSerializersRegistrar.scala:22]: Maybe let's also extract it to a method?

File: engine/flink/generic/src/main/resources/defaultModelConfig.conf (hunk 2 above)
Comment (mproch) [reply engine/flink/generic/src/main/resources/defaultModelConfig.conf:16]: it should... I also wonder about consumerGroupNamingStrategy - shouldn't it be the default?

File: engine/flink/management/sample/src/main/resources/defaultModelConfig.conf [hunk 4]
```diff
@@ -48,6 +48,10 @@ componentsUiConfig: {
 }
 
 components {
+  kafka {
+    disabled: true
+  }
```
Comment (mproch) [inline engine/flink/management/sample/src/main/resources/defaultModelConfig.conf:53]: Do we need it if it's not autoloaded?

File: components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/KafkaComponentsProvider.scala [hunk 5]
```diff
@@ -24,19 +26,23 @@ class KafkaComponentsProvider extends ComponentProvider {
...
-    ComponentDefinition("kafka-registry-typed-json", new KafkaAvroSourceFactory(jsonSerializingSchemaRegistryProvider, dependencies, None)),
-    ComponentDefinition("kafka-registry-typed-json", new KafkaAvroSinkFactoryWithEditor(jsonSerializingSchemaRegistryProvider, dependencies)),
-    ComponentDefinition("kafka-registry-typed-json-raw", new KafkaAvroSinkFactory(jsonSerializingSchemaRegistryProvider, dependencies)),
-    ComponentDefinition("kafka-avro-raw", new KafkaAvroSinkFactory(avroSerializingSchemaRegistryProvider, dependencies)),
-  )
+  override def create(config: Config, dependencies: ProcessObjectDependencies): List[ComponentDefinition] = {
+    val kafkaConfig = config.getConfig("config")
+    val kafkaConfigMergedWithGlobalConfig = dependencies.config.withValue("kafka", fromAnyRef(kafkaConfig.root()))
+    List(
+      ComponentDefinition("kafka-json", new GenericKafkaJsonSinkFactory(dependencies.copy(config = kafkaConfigMergedWithGlobalConfig))),
```
[the remark this answers is not in the document]
Comment (dswiecki) [reply components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/KafkaComponentsProvider.scala]: done

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/kryo/AvroSerializersRegistrar.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (dswiecki) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/kryo/AvroSerializersRegistrar.scala]: done

File: engine/flink/generic/src/main/resources/defaultModelConfig.conf (hunk 2 above)
Comment (dswiecki) [reply engine/flink/generic/src/main/resources/defaultModelConfig.conf:16]: done

File: engine/flink/management/sample/src/main/resources/defaultModelConfig.conf (hunk 4 above)
Comment (dswiecki) [reply engine/flink/management/sample/src/main/resources/defaultModelConfig.conf]: done

File: nussknacker-dist/src/universal/conf/application.conf [hunk 6]
```diff
@@ -40,3 +40,12 @@ countsSettings {
   influxUrl: ${?COUNTS_URL}
   database: "esp"
 }
+
+kafka {
+  kafkaAddress: ${?KAFKA_ADDRESS}
+  kafkaProperties {
+    "schema.registry.url": ${?SCHEMA_REGISTRY_URL}
+  }
+  consumerGroupNamingStrategy: processId-nodeId
```
Comment (arkadius) [inline nussknacker-dist/src/universal/conf/application.conf:49]: it is default strategy - you can remove it. this application.conf should be as smallest as possible

File: nussknacker-dist/src/universal/conf/application.conf [hunk 7]
```diff
@@ -40,3 +40,12 @@ countsSettings {
...
+  avroKryoGenericRecordSchemaIdSerialization: true
```
Comment (arkadius) [inline nussknacker-dist/src/universal/conf/application.conf:50]: Why we have two properties: avroKryoGenericRecordSchemaIdSerialization and autoRegisterRecordSchemaIdSerialization? And why it is here if default is true?

Comment (dswiecki) [reply nussknacker-dist/src/universal/conf/application.conf]: i've change default if config not provided

File: nussknacker-dist/src/universal/conf/application.conf (hunk 6 above)
Comment (dswiecki) [reply nussknacker-dist/src/universal/conf/application.conf]: removed from here

**Review by arkadius — APPROVED** [review]

**Pull request merged**